lib/moonshot/dynamic_template.rb

```diff
@@ -58,9 +58,37 @@ def validate_destination_exists
     end
 
     def validate_template(template)
-      Aws::CloudFormation::Client.new.validate_template(
-        template_body: template
-      )
+      if template.bytesize > 50_000 # Leave some margin from the 51,200 limit
+        s3_client = Aws::S3::Client.new
+        bucket_name = "#{Moonshot.config.template_s3_bucket}"
```

Comment:
if this returns empty or nil because template_s3_bucket we should raise an error